src/CommunityToolkit/Xamarin.CommunityToolkit/Core/StreamMediaSource.shared.cs

```diff
@@ -47,12 +47,16 @@ protected override void OnPropertyChanged(string propertyName)
 			base.OnPropertyChanged(propertyName);
 		}
 
-		async Task<Stream> IStreamImageSource.GetStreamAsync(CancellationToken userToken)
+		async Task<Stream?> IStreamImageSource.GetStreamAsync(CancellationToken userToken)
 		{
 			if (Stream == null)
 				return null;
 
 			OnLoadingStarted();
+
+			if (CancellationTokenSource == null)
+				throw new Exception($"{nameof(OnLoadingStarted)} not called");
```

Comment:
Maybe `InvalidOperationException`, or something like that. It's not recommended to throw `Exception`. You'll even get a warning if you try to write `catch (Exception ex)`